firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1844,63 +1844,120 @@ class BrowserViewController: UIViewController,
         }
     }
 
-    private func creditCardAutofillSetup(_ tab: Tab, didCreateWebView webView: WKWebView) {
+    /// Returns whether the credit card autofill feature flag is enabled from Nimbus
+    func autofillCreditCardNimbusFeatureFlag() -> Bool {
+        return featureFlags.isFeatureEnabled(.creditCardAutofillStatus, checking: .buildOnly)
+    }
+
+    /// Returns whether credit card autofill settings are enabled based on user defaults key KeyAutofillCreditCardStatus
+    func autofillCreditCardSettingsUserDefaultIsEnabled() -> Bool {
         let userDefaults = UserDefaults.standard
         let keyCreditCardAutofill = PrefsKeys.KeyAutofillCreditCardStatus
 
-        let autofillCreditCardStatus = featureFlags.isFeatureEnabled(
-            .creditCardAutofillStatus, checking: .buildOnly)
+        return (userDefaults.object(forKey: keyCreditCardAutofill) as? Bool ?? true)
+    }
+
+    /// Returns whether the address autofill feature flag is enabled from Nimbus
+    func addressAutofillNimbusFeatureFlag() -> Bool {
+        return featureFlags.isFeatureEnabled(.addressAutofill, checking: .buildOnly)
+    }
+
+    /// Returns whether address autofill settings are enabled based on user defaults.
+    func addressAutofillSettingsUserDefaultsIsEnabled() -> Bool {
+        let userDefaults = UserDefaults.standard
+        let keyAddressAutofill = PrefsKeys.KeyAutofillAddressStatus
+
+        return (userDefaults.object(forKey: keyAddressAutofill) as? Bool ?? true)
+    }
+
+    /// Sets up credit card autofill handling.
+    private func creditCardAutofillSetup(_ tab: Tab, didCreateWebView webView: WKWebView) {
         let formAutofillHelper = FormAutofillHelper(tab: tab)
         tab.addContentScript(formAutofillHelper, name: FormAutofillHelper.name())
+
+        // Closure to handle found field values for credit card and address fields
         formAutofillHelper.foundFieldValues = { [weak self] fieldValues, type, frame in
-            guard let tabWebView = tab.webView,
-                  let type = type,
-                  userDefaults.object(forKey: keyCreditCardAutofill) as? Bool ?? true
-            else { return }
+            guard let strongSelf = self, let tabWebView = tab.webView else { return }
```

Comment:
nit: you don't actually need strongSelf but instead you can do the following according to [SE-0365](https://github.com/apple/swift-evolution/blob/main/proposals/0365-implicit-self-weak-capture.md)
```suggestion
            guard let self, let tabWebView = tab.webView else { return }
```